### PR TITLE
[BUGFIX] Fix RegexPatternStringParameterBuilder to use legal character repetition.  Remove median, mean, and standard deviation features from OnboardingDataAssistant "datetime_columns_rule" definition.

### PR DESCRIPTION
**Pull request by alexsherstinsky**
### Scope
* Legal limit for character repetition in Regular Expressions is `255` (not `256`).
* Delete unsupported statistics from the datetime_columns_rule in OnboardingDataAssistant (e.g., PosgreSQL does not support the `AVERAGE` operation for `DATETIME` columns.

Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

Changes proposed in this pull request:
- JIRA: GREAT-761/GREAT-1176
- JIRA: GREAT-761/GREAT-1181
-

After submitting your PR, CI checks will run and @cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

**Review by NathanFarmer — APPROVED** [review]
Looks good to me, assuming it passes tests.

File: great_expectations/execution_engine/execution_engine.py [hunk 1]
```diff
@@ -408,7 +408,8 @@ def resolve_metrics(
                 resolved_metrics.update(new_resolved)
             except Exception as e:
                 raise ge_exceptions.MetricResolutionError(
-                    message=str(e), failed_metrics=[x[0] for x in metric_fn_bundle]
+                    message=str(e),
+                    failed_metrics=[x.metric_configuration for x in metric_fn_bundle],
```
Comment (Shinnnyshinshin) [inline great_expectations/execution_engine/execution_engine.py:412]: 🙇🏼

File: great_expectations/rule_based_profiler/data_assistant/onboarding_data_assistant.py [hunk 2]
```diff
@@ -626,15 +626,6 @@ def _build_datetime_columns_rule() -> Rule:
...
         )
-        column_median_metric_multi_batch_parameter_builder_for_metrics: ParameterBuilder = (
-            DataAssistant.commonly_used_parameter_builders.get_column_median_metric_multi_batch_parameter_builder()
-        )
-        column_mean_metric_multi_batch_parameter_builder_for_metrics: ParameterBuilder = (
-            DataAssistant.commonly_used_parameter_builders.get_column_mean_metric_multi_batch_parameter_builder()
-        )
-        column_standard_deviation_metric_multi_batch_parameter_builder_for_metrics: ParameterBuilder = (
-            DataAssistant.commonly_used_parameter_builders.get_column_standard_deviation_metric_multi_batch_parameter_builder()
-        )
```
Comment (Shinnnyshinshin) [inline great_expectations/rule_based_profiler/data_assistant/onboarding_data_assistant.py:637]: ❤️ and the more I think of it, the more I think this makes sense

**Review by Shinnnyshinshin — APPROVED** [review]
Thank you very much for this @alexsherstinsky. A big unlock 🔑 🔒

**Pull request merged**